pkg/chains/signing/kms/kms.go

```diff
@@ -120,6 +137,28 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
 	}, nil
 }
 
+// getKMSAuthToken retreives token from the given mount path
+func getKMSAuthToken(dir string) (string, error) {
+	tokenEnv := "KMS_AUTH_TOKEN"
+	stat, err := os.Stat(dir)
+	if err != nil {
+		return "", err
+	}
+	// If the path exists but is not a directory, then throw an error
+	if !stat.IsDir() {
+		return "", fmt.Errorf("path specified %s is not a directory", dir)
+	}
+
+	filePath := filepath.Join(dir, tokenEnv)
+	fileData, err := os.ReadFile(filePath)
+	if err != nil {
+		return "", err
```

Comment:
Maybe wrap this error to make it clear which file Chains is trying to load:

```go
return "", fmt.Errorf("reading file %q in directory %q: %w", tokenEnv, dir, err)
```

Reply:
Incorporated